lib/corefoundation/preferences.rb

```diff
@@ -23,49 +20,133 @@ module CF
   attach_function "CFPreferencesSetValue", %i{key value application_id username hostname}, :void
   attach_function "CFPreferencesAppSynchronize", [:application_id], :bool
 
+  # Interface to the preference utilities from Corefoundation.framework.
+  # Documentation at https://developer.apple.com/documentation/corefoundation/preferences_utilities
+  #
   class Preferences
     CURRENT_USER = CF.kCFPreferencesCurrentUser
     ALL_USERS = CF.kCFPreferencesAnyUser
     CURRENT_HOST = CF.kCFPreferencesCurrentHost
     ALL_HOSTS = CF.kCFPreferencesAnyHost
 
-    class << self
```

Comment:
Since most methods are class, let's continue using this to reduce noise in method declarations

Reply:
Doesn't it make it easier if the method name also has the hint if it is a class method or not? Otherwise you would rely on checking if it is inside a nested block (another reason to avoid this?). IMO a second if not two well saved

Reply:
Refer rubocop style guide, though we have disabled some cops, still good enough for common ground, https://github.com/rubocop/ruby-style-guide#def-self-class-methods

Reply:
In the link you provided both `self.` and `<<` as mentioned as a good way. I scoured the internet quite a bit and found out that usage of `self` or `<<` mostly depends on individual teams. So rubocop doesn't have anything against it. 😄

Reply:
I did my due diligence before making the change haha

Reply:
Read that reference doc/example again and let me know if you still can't find it. Read some ruby code (OSS) as well, sinatra, rails, puma, sidekiq etc.

Reply:
I have. I'm not sure what leads you to believe that rubocop is suggesting one over the other.

Reply:
Quoting the rubocop doc, there is more if you need other references. Also, this is preferred by rubyists I have interacted over work/meetups.

```ruby
# Also possible and convenient when you
# have to define many class methods.
  class << self
    def first_method
      # body omitted
    end

    def second_method_etc
      # body omitted
    end
  end
 ```

Reply:
All I can see is that it says it is possible and a convenient alternative. Doesn't say it should be preferred. Which is why I was confused why you think it is suggesting to use one over the other. And speaking of other people's opinions, here's something you can take a look at: https://stackoverflow.com/questions/10964081/class-self-vs-self-method-with-ruby-whats-better. (Please take the time to read through all of the comments)

Reply:
Bottom line is, I would rather prefer reasoning over following the herd. And hopefully this will put an end to this rather silly conversation, here are my reasons for why I'm going with the `self.method_name` for this.

1. `self.method_name` seems to a more idiomatic way of writing it. Meaning, you see a method and you know what it is. You don't have to scroll and figure out if it is inside the block.
2. I would prefer using something that doesn't introduce a particular behaviour, unless that behaviour is needed - This refers to how `class << self` notation works. Since it opens the eigenclass, defining constants inside that block wouldn't be available within the main class. This won't happen without it. This is only functional difference between using `class << self` and `self.method_name` notation. So unless we need that behaviour, I would prefer sticking to `self.method_name`